### ISSUE TITLE
Some questions that occured to me while reading this spec

### ISSUE THREAD

**Issue by Gozala** [RhizomeDB/pomo-ipld#2]
1. IPLD schema says that entity id is bytes, but then int is used to in the examples. This is really confusing and I'm not sure which one out of two is correct

https://github.com/RhizomeDB/pomo-ipld/blob/65484801ef5159ae42f701895ffee25e88ebd7ef/README.md?plain=1#L51-L64

2. Could you please provide some rational for the 128bit limit rational ? I'm under impression that perhaps use ofy [u128](https://doc.rust-lang.org/std/primitive.u128.html#) was intention which might explain my first question

https://github.com/RhizomeDB/pomo-ipld/blob/65484801ef5159ae42f701895ffee25e88ebd7ef/README.md?plain=1#L68

3. Should `CausedBy` canonicalization be prescribed, or better yet be defined as a `Map<CIDBytes, Unit>` to ensure that same fact will hash the same ?

https://github.com/RhizomeDB/pomo-ipld/blob/65484801ef5159ae42f701895ffee25e88ebd7ef/README.md?plain=1#L105-L109

4. I think something is missing in this sentence, is trivial to handle ? 

> the same fact entered into the store twice is trivial for operations that only depend on the graph structure of the store

5. Perhaps it's better idea to recommend using multihashes instead ? IPFS stack through experience settled on using multihashes (stripping codec and cid tags) for block stores as that information usually does not affect most operations. In case of DB it makes little sense IMO to have even hashing algorithm retained, individual DB could probably choose whatever algorithm it cares and use untagged `digests` for keys. When communicating with others DB could choose to add prefixes etc...

https://github.com/RhizomeDB/pomo-ipld/blob/65484801ef5159ae42f701895ffee25e88ebd7ef/README.md?plain=1#L145-L153

**Comment by Gozala** [comment]
I also have been talking to @expede on discord about what were limits for `causedBy` and how to handle cases where you may need to have more than that number. I think spec should probably cover that

**Comment by expede** [comment]
Yep agreed! The spec is very WIP and Fission has it on hold. PRs welcome :)

**Comment by Gozala** [comment]
I'm interested in contributing, having above questions answered would enable me to make align with the vision.
